### ISSUE TITLE
Lint for setting advanceTimers with user-event when using jest.useFakeTimers

### ISSUE THREAD

**Issue by backwardok** [testing-library/eslint-plugin-testing-library#979]
### Name for new rule

advance-timers-with-fake-timers

### Description of the new rule

Enforce using the `advanceTimers` option for `user-event` when using jest's `useFakeTimers` in order to prevent testing timeouts

### Testing Library feature

[user-event's advanceTimers option](https://testing-library.com/docs/user-event/options/#advancetimers) when [using fake timers](https://testing-library.com/docs/using-fake-timers)

### Testing Library framework(s)

user-event (across any Testing Library framework that can use it)

### What category of rule is this?

Warns about a potential error

### Code examples

Using fake timers within a test directly:
```js
jest.useFakeTimers();
const user = userEvent.setup(); // Warn - advanceTimers not set
```

```js
beforeEach(() => {
  jest.useFakeTimers()
});

test('some behavior with user action', async () => {
  const user = userEvent.setup(); // Warn - advanceTimers not set
  // Test behavior
})
```

### Do you want to submit a pull request to make the new rule?

No

**Comment by Belco90** [comment]
Hi @backwardok. Thanks for your suggestion, sounds like an interesting rule!

**Comment by alexeyr-ci2** [comment]
Should it also warn when using `userEvent.click(...)` and similar methods directly without `userEvent.setup()`?

**Comment by Belco90** [comment]
> Should it also warn when using `userEvent.click(...)` and similar methods directly without `userEvent.setup()`?

The Testing Library docs only reference passing the `advanceTimers` to the `setup` method, so I don't think other similar methods should be reported.
